nixos/modules/programs/gpu-screen-recorder.nix

```diff
@@ -34,6 +40,36 @@ in {
       capabilities = "cap_sys_nice+ep";
       source = "${package}/bin/gpu-screen-recorder";
     };
+    systemd.services.gpu-screen-recorder = {
+      description = "GPU Screen Recorder Service";
+      after = [ "network.target" ];
+      serviceConfig = {
+        EnvironmentFile = "-%h/.config/gpu-screen-recorder.env";
+        ExecStart = ''
+          ${lib.getExe' pkgs.gpu-screen-recorder "gpu-screen-recorder"} \
+            -v no \
+            -w "''${WINDOW:-screen}" \
+            -c "''${CONTAINER:-mp4}" \
+            -q "''${QUALITY:-very_high}" \
+            -k "''${CODEC:-auto}" \
+            -ac "''${AUDIO_CODEC:-opus}" \
+            -a "''${AUDIO_DEVICE:-default_output}" \
+            -f "''${FRAMERATE:-60}" \
+            -r "''${REPLAYDURATION:-30}" \
+            -o "''${OUTPUTDIR:-%h/Videos}" \
+            -df "''${MAKEFOLDERS:-no}" \
+            $ADDITIONAL_ARGS \
+            -cr "''${COLOR_RANGE:-limited}" \
+            -keyint "''${KEYINT:-2}" \
+            -restore-portal-session "''${RESTORE_PORTAL_SESSION:-yes}" \
+            -encoder "''${ENCODER:-gpu}"
```

Comment:
From my perspective, these should all be options in the module. Something like `programs.gpu-screen-recorder.service.replayDuration`. dec05eba may feel differently.

Reply:
Using nix program options sounds better yeah. The systemd service file is simple and its more of an example of how to start recording at system startup.

Reply:
I will work on this in another PR

https://github.com/NixOS/nixpkgs/pull/339874#issuecomment-2395418949